### ISSUE TITLE
Proposal: Ignore refactoring opportunities in pipe chain when using erlang functions

### ISSUE THREAD

**Issue by maisonbleu** [rrrene/credo#431]
### Environment

* Credo version (`mix credo -v`): 0.8.6
* Erlang/Elixir version (`elixir -v`): Elixir 1.5.1 and Erlang/OTP 20
* Operating system: macOS 10.12

### What were you trying to do?

:crypto.hash(:md5, "test") |> Base.encode16(case: :lower)

### Expected outcome

Expected credo to ignore piping for erlang functions.

### Actual outcome

Refactoring opportunities: _Pipe chain should start with a raw value._

In my opinion it doesn't make sense to report a refactoring opportunity in a pipe chain with erlang modules as they are not created with piping in mind.

Best wishes,
Patrick

**Comment by rrrene** [comment]
@maisonbleu  I can see your argument and would probably accept a PR adding a config param for this (which defaults to true).

FWIW: You can exclude functions from the check

```elixir
{Credo.Check.Refactor.PipeChainStart, excluded_functions: ~w(:crypto.hash)},
```

**Comment by stefanchrobot** [comment]
I'd like to suggest to extend the scope of this item. I think the check is great when applied to compound values, like the one in the example in the explanation:
```
list
|> Enum.take(5)
|> Enum.shuffle
|> pick_winner()
```
But I think this does not work, when the value that starts the chain is simple (number, atom, maybe string). This:
```
:crypto.strong_rand_bytes(14)
|> Base.url_encode64()
|> binary_part(0, 14)
```
seems to read better than:
```
14
|> :crypto.strong_rand_bytes()
|> Base.url_encode64()
|> binary_part(0, 14)
```
Similar case for ExMachina's factories:
```
build(:user)
|> to_json!()
|> Map.put_new("password", "secret")
|> Map.merge(attrs)
```
I'm on the fence with Ecto queries, though:
```
Repo.get_by(User, email: String.downcase(email))
|> check_password(password)
```

Adding a flag to ignore this rule for atoms, numbers and strings would solve those cases and the original reported case too.

**Comment by asummers** [comment]
Just chiming in to say I think that the fix here should be an option rather than the only behavior. I can see the argument for all of these things, but fall on the side of the existing behavior for all of them.

**Comment by eagle-te** [comment]
@rrrene 
Hi, I'm in need of some enlightenment. I'm trying to add a function to the `excluded_functions` section of `PipeChainStart`, but non of the variation I tried seem to work: 
```elixir
excluded_functions: ["function_name",~w(function_name),:function_name, ~w(:function_name), "Module.Name.function_name"]
```
Credo still shows it as refactoring opportunity.
I've the feeling, that I'm doing something wrong.
